bitcampHomework21/src/ba/bitcamp/homework21/tasks/Client.java

```diff
@@ -0,0 +1,179 @@
+import java.awt.Desktop;
+import java.awt.Font;
+import java.awt.GridLayout;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+
+import java.net.Socket;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.UnknownHostException;
+
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.JTextArea;
+import javax.swing.JTextField;
+
+/**
+ * This class is used as client, that will communicate with server using sockets
+ * ports and ip addresses. It simulates GUI chat application
+ * 
+ * @author Mladen13
+ *
+ */
+public class Client extends JFrame {
+
+	private static final long serialVersionUID = 1298383866907516111L;
+	// GUI components
+	private JButton send = new JButton("Send");
+	private JTextArea area = new JTextArea();
+	private JTextField field = new JTextField();
+	private JPanel panel = new JPanel();
+	private JPanel panel2 = new JPanel();
+	// Client components
+	Socket client = null;
+	BufferedWriter writer = null;
+	BufferedReader reader = null;
+	String s = "";
+
+	/**
+	 * Constructor that creates client and gui application with components on
+	 * frame
+	 * 
+	 * @throws UnknownHostException
+	 * @throws IOException
+	 */
+	public Client() throws UnknownHostException, IOException {
+
+		field.addActionListener(new ButtonHandler());
+		send.addActionListener(new ButtonHandler());
+
+		setTitle("Client");
+		setSize(500, 500);
+		setLocationRelativeTo(null);
+		setDefaultCloseOperation(EXIT_ON_CLOSE);
+		// Creating client
+		client = new Socket("192.168.0.100", 8888);
+		// Adding gui components to frame
+		setLayout(new BorderLayout());
+		add(panel, BorderLayout.CENTER);
+		add(panel2, BorderLayout.SOUTH);
+		add(field, BorderLayout.NORTH);
+		// adding text area to panel
+		panel.setLayout(new GridLayout(1, 1));
+		panel.add(area);
+		// setting text area look
+		area.setEnabled(false);
+		area.setBackground(Color.GRAY);
+		area.setOpaque(true);
+		area.setFont(new Font("Monospaced", Font.BOLD, 15));
+		// adding components to second panel
+		panel2.setLayout(new GridLayout(1, 1));
+		panel2.add(send);
+
+		setVisible(true);
+		// Creating buffered writer and reader.
+		reader = new BufferedReader(new InputStreamReader(
+				client.getInputStream()));
+		writer = new BufferedWriter(new OutputStreamWriter(
+				client.getOutputStream()));
+		// Reading lines from client and appending them to text area
+		while ((s = reader.readLine()) != null) {
+			if (s.length() > 0) {
+
+				checkLine(s);
+			}
+		}
+	}
+
+	/**
+	 * This method checks if our line contains some of wanted inputs
+	 * 
+	 * @param s
+	 *            - represents one line
+	 * @throws IOException
+	 */
+	public void checkLine(String s) throws IOException {
+		if (s.contains("/web ")) {
```

Comment:
Dobar "fazon" :smile:  za ove slucajeve je da napises liniju kao: 

    } else if ("/web".equals(s)) {

jer se moze desiti da je `s` = `null` pa ce biti `NullException` ovako to se nece nikad desiti.